pkg/handler/repositories.go

```diff
@@ -68,24 +71,68 @@ func listRepositories(c echo.Context) error {
 func createRepository(c echo.Context) error {
 	newRepository := api.CreateRepository{}
 	if err := c.Bind(&newRepository); err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, "error binding params: "+err.Error())
+		return echo.NewHTTPError(http.StatusBadRequest, "Error binding params: "+err.Error())
 	}
 
 	AccountID, OrgID, err := getAccountIdOrgId(c)
 	if err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, "error parsing identity: "+err.Error())
+		return echo.NewHTTPError(http.StatusBadRequest, "Error parsing identity: "+err.Error())
 	}
 	newRepository.AccountID = AccountID
 	newRepository.OrgID = OrgID
 
 	repositoryDao := dao.GetRepositoryDao()
 	if err := repositoryDao.Create(newRepository); err != nil {
-		return echo.NewHTTPError(http.StatusInternalServerError, "error creating repository: "+err.Error())
+		return echo.NewHTTPError(codeForError(err), "Error creating repository: "+err.Error())
 	}
 
 	return c.String(http.StatusCreated, "Repository created.\n")
 }
 
+// CreateRepository godoc
+// @Summary      Update Repository
+// @ID           updateRepository
+// @Description  Update a repository
+// @Tags         repositories
+// @Accept       json
+// @Produce      json
+// @Param  body       body    api.CreateRepository true  "request body"
+// @Param  org_id     header  string         	   true  "organization id"
+// @Param  account_id header  string               true  "account number"
+// @Success      201
+// @Router       /repositories/{uuid} [put]
+func fullUpdate(c echo.Context) error {
+	uuid := c.Param("uuid")
+	repoParams := api.CreateRepository{}
+	_, OrgID, err := getAccountIdOrgId(c)
+	dao := dao.GetRepositoryDao()
+
+	if err := c.Bind(&repoParams); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "error binding params: "+err.Error())
+	}
+
+	err = dao.FullUpdate(OrgID, uuid, repoParams)
+	if err != nil {
+		return echo.NewHTTPError(codeForError(err), err.Error())
+	}
+	return c.String(http.StatusOK, "Repository Updated.\n")
+}
+
+func codeForError(err error) int {
```

Comment:
what about naming this httpCodeForError, httpErrorCode, or something with "http" in it?